CMakeLists.txt

```diff
@@ -164,6 +164,46 @@ if ( (NOT MPI_C_FOUND) OR (NOT MPI_Fortran_FOUND) )
   find_package( MPI REQUIRED )
 endif()
 
+# Test for consistent MPI environment
+if (NOT MPIEXEC)
+  message ( ERROR "CMake failed to find `mpiexec` or similar. If building with `./install.sh` please
+report this bug to the OpenCoarrays developers at
```

Comment:
This should throw an error if MPIEXEC is not found, and catch #359 or similar in the future